sigmf/utils.py

```diff
@@ -91,9 +91,13 @@ def get_endian_str(ray):
         # endianness must be either '=' (native) or '|' (doesn't matter)
         return '_le' if sys.byteorder == 'little' else '_be'
 
-def get_data_type_str(ray):
+def get_data_type_str(ray, interleaved_complex_int = False):
```

Comment:
Not sure I'm in love with this kwarg, but I don't see an obvious alternative. Also there shouldn't be spaces around your `=`.

Reply:
I'll fix the whitespace thanks. I'm also not in love with this.... I debated leaving this out all together however after internal debate I think `ci16` is common enough to warrant it. Open to removal if there are strong opinions.

Reply:
Yea we have to support `ci16`, but in my mind the whole point of calling the `get` is to resolve the ambiguity with the datatype. If it's not possible to determine with numpy, perhaps it should return some ambiguity like a tuple of possibilities or an error saying it cannot determine if it's `ci16` for `f8` or something.

Reply:
@gmabey the issue comes about because the canonical numpy representation of ci16 data *is* as a np.int16 array.

Reply:
@jacobagilbert <sigh> maybe I should back up.  I work with `ci16` data every day, just like you.  That's why the `__getitem__()` method in #209 works with this datatype.

I think that have already dealt (very much) with the issues you and @Teque5 are discussing.  The difference is that I'm not just unhappy with `interleaved_complex_int`, I'm trying to say that it seems to be an attempt to turn `get_data_type_str()` into a universal datatype string generator, which was not my intent.

I feel that returning a tuple of possibly-compatible-but-not-equivalent-to-ray.dtype strings is an awful complication to what *was* supposed to be a straightforward *convenience* function.

Also, because I first created a new comment associated with the PR instead of replying to this one (like I should have done, sorry -- and I didn't flag you) you might have missed part of this conversation (maybe).

Reply:
I also do not think ambiguous returns are what we want.

The `get_data_type_str()` function is being interpreted as a datatype string generator because the docstring says

> Return the SigMF datatype string for the datatype of numpy array `ray`.

One option is to update the docstring to indicate that this function does not work for ci16_* data, but something needs to address this part of the problem while we fix the other bug in this. In general, convenience functions should not increase edge cases, which leads me to the `cplx_interleaved_int` arg... still leaning that direction of the various not-so-god options we have.

Reply:
Sorry for not following back up with this sooner, but I think that docstring is perfectly accurate as-is, since it's not possible to have a `numpy` array of type `ci16`.  So, anything that you *can* pass to `get_data_type_str()` will result in the correct SigMF typestring being returned.  But all that aside -- thank you for being willing to accept the lowest common denominator of consensus.